pkg/provider/aws/infra_template.go

```diff
@@ -172,3 +168,40 @@ func (a *AWS) Transform(result *core.CompilationResult, deps *core.Dependencies)
 	result.Add(data)
 	return errs.ErrOrNil()
 }
+
+func (a *AWS) GenerateCloudfrontDistributions(data *TemplateData, result *core.CompilationResult) {
+	cloudfrontMap := make(map[string][]core.CloudResource)
+	for _, res := range result.Resources() {
+		key := res.Key()
+		switch res.(type) {
+		case *core.Gateway:
```

Comment:
nit: i think aside from the `a.Config.GetExposed(key.Name)` and `a.Config.GetStaticUnit(key.Name)` the rest of the logic is the same so the switch could just set that and the rest of the logic moved out and de-duped